client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+        }
+
+        //verify the redirect request has the same Authorization header
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
+    }
+
+    @Test
+    public void testClientRedirectEnabled()
+            throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults(server)));
+
+        QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.of(false)), createClientSession(server));
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+        }
+
+        //verify the redirect request has the same Authorization header
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
+    }
+
+    @Test
+    public void testClientRedirectDisabled()
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect_host/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        try {
+            QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.of(true)), createClientSession(server));
+
+            try (Query query = queryRunner.startQuery("SELECT 1")) {
+                query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+            }
+        }
+        catch (Exception e) {
+            assertThat(e).isInstanceOf(ClientException.class);
+            assertThat(e.getMessage()).isEqualTo("Client redirection was detected, but 'disableFollowRedirects' flag is set to true");
+        }
+    }
+
+    static TrinoUri createTrinoUri(MockWebServer server, Optional<Boolean> disableFollowRedirects)
+            throws SQLException
+    {
+        Properties properties = new Properties();
+        if (disableFollowRedirects.isPresent()) {
+            properties.setProperty(PropertyName.DISABLE_FOLLOW_REDIRECTS.toString(), Boolean.toString(disableFollowRedirects.get()));
+        }
+        properties.setProperty(PropertyName.ACCESS_TOKEN.toString(), DUMMY_ACCESS_TOKEN);
```

Comment:
Test the following:
password (`Authorization: Basic <credentials>`)
accessToken, externalAuthentication (`Authorization: Bearer <token>`)

Network interceptor for password is missing in the current implementation.

Reply:
Added test cases for `basicAuth`, `accessToken` and `externalAuth`